dev/com.ibm.ws.security.wim.adapter.ldap_fat/fat/src/com/ibm/ws/security/wim/adapter/ldap/fat/LDAPMemberAttributeScopeTest.java

```diff
@@ -0,0 +1,577 @@
+
+        // TODO group_level1 is returned b/c recursiveSearch=true overrides search scope in LdapAdapter.getGroupsByMember()
+        results = urServlet.getGroupsForUser(MEMBER_DYNAMIC_USER_DN);
+        assertMembershipResults("With all scope, the dynamic member should be a member of the dynamic group.", results, MEMBER_LEVEL1_GROUP_DN, MEMBER_DYNAMIC_GROUP_DN);
+
+        // TODO user_level1 is returned b/c recursiveSearch=true overrides search scope in LdapAdapter.getMembersByMember()
+        results = urServlet.getUsersForGroup(MEMBER_LEVEL1_GROUP_DN, 0).getList();
+        assertMembershipResults("With all scope, the root group should have the direct member.", results, MEMBER_LEVEL1_USER_DN, MEMBER_LEVEL2_USER_DN);
+
+        results = urServlet.getUsersForGroup(MEMBER_LEVEL2_GROUP_DN, 0).getList();
+        assertMembershipResults("With all scope, the nested group should have only the nested member.", results, MEMBER_LEVEL2_USER_DN);
+
+        results = urServlet.getUsersForGroup(MEMBER_DYNAMIC_GROUP_DN, 0).getList();
+        assertMembershipResults("With all scope, the dynamic group should have only the dynamic member.", results, MEMBER_DYNAMIC_USER_DN);
+    }
+
+    /**
+     * Test a membershipAttribute whose scope is set to "direct". Direct scoped membership attributes
+     * only return their direct memberships, thus this will require nested calls to derive
+     * group membership.
+     *
+     * @throws Exception if the test failed for some unforeseen reason.
+     */
+    @Test
+    public void membershipAttribute_scope_direct() throws Exception {
+
+        ServerConfiguration clone = baseConfiguration.clone();
+        MembershipAttribute membershipAttr = new MembershipAttribute("memberof", LdapConstants.LDAP_DIRECT_GROUP_MEMBERSHIP_STRING);
+        createLdapRegistry(clone, MEMBERSHIP_BASE_DN, null, membershipAttr);
+        LDAPFatUtils.updateConfigDynamically(libertyServer, clone);
+
+        List<String> results = urServlet.getGroupsForUser(MEMBERSHIP_LEVEL1_USER_DN);
+        assertMembershipResults("With direct scope, the direct member should only be a member of the root group.", results, MEMBERSHIP_LEVEL1_GROUP_DN);
+
+        results = urServlet.getGroupsForUser(MEMBERSHIP_LEVEL2_USER_DN);
+        assertMembershipResults("With direct scope, the nested member should be a member of the root and nested group.", results, MEMBERSHIP_LEVEL1_GROUP_DN,
+                                MEMBERSHIP_LEVEL2_GROUP_DN);
+
+        results = urServlet.getGroupsForUser(MEMBERSHIP_DYNAMIC_USER_DN);
+        assertMembershipResults("With direct scope, the dynamic member should be a member of the root and dynamic group.", results, MEMBERSHIP_DYNAMIC_GROUP_DN);
+
+        results = urServlet.getUsersForGroup(MEMBERSHIP_LEVEL1_GROUP_DN, 0).getList();
+        assertMembershipResults("With direct scope, the root group should have both the direct and nested members.", results, MEMBERSHIP_LEVEL1_USER_DN, MEMBERSHIP_LEVEL2_USER_DN);
```

Comment:
The dynamic user should be a member of the level 1 group, but the LdapAdapter.getMembersByMembership() method does not consider nested dynamic groups.

Reply:
Opened issue #14249.